src/main/java/com/learning/browsers/BrowserDriverGenerator.java

```diff
@@ -0,0 +1,30 @@
+package com.learning.browsers;
+
+import com.learning.configuration.PropertiesReader;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.edge.EdgeDriver;
+
+public class BrowserDriverGenerator implements BrowserDriver {
+
+  private static final String BROWSER_PROPERTIES_NAME = "browser";
+  private static final String PATH_TO_EDGE_DRIVER = "pathToEdgeDriver";
+  private final PropertiesReader propertiesReader;
+
+  public BrowserDriverGenerator() {
+    propertiesReader = new PropertiesReader();
+  }
+
+  public WebDriver getDriver() {
+    String browserName = propertiesReader.getProperty(BROWSER_PROPERTIES_NAME);
+    if (browserName.equalsIgnoreCase("Chrome")) {
+      return new ChromeDriver();
+    } else if (browserName.equalsIgnoreCase("Edge")) {
+      System.setProperty("webdriver.edge.driver", propertiesReader.getProperty(PATH_TO_EDGE_DRIVER));
+      return new EdgeDriver();
```

Comment:
I am not a big fan of Factory Pattern since I think about it like pseudo object oriented. 
Could you suggest some more elegant solution for building WebDriver?